queue/src/main/java/org/killbill/notificationq/DefaultNotificationQueue.java

```diff
@@ -327,6 +337,44 @@ public Void withSqlDao(final NotificationSqlDao transactional) throws Exception
         InTransaction.execute(dbi, connection, handler, NotificationSqlDao.class);
     }
 
+    @Override
+    public void removeFutureNotificationsForSearchKeys(final Long searchKey1, final Long searchKey2) {
+        dao.getSqlDao().inTransaction(new Transaction<Void, QueueSqlDao<NotificationEventModelDao>>() {
+            @Override
+            public Void inTransaction(final QueueSqlDao<NotificationEventModelDao> transactional, final TransactionStatus status) throws Exception {
+                // Move entries by batch into the history table
+                final int batchSize = 25;
```

Comment:
Should we return an indicator to the user saying that there are more entries than the default batch size, or if not, how will she know that she should call back the api?

Reply:
All entries would be removed, this is just to call `dao.moveEntriesToHistoryFromTransaction` by batch (`@SqlBatch`), instead of one at a time. See https://github.com/killbill/killbill-commons/pull/45/files/c2b601f7093ff881a9cd3b7efe029f4820208335#diff-0b9d631efc836141ed593b2514afb5e8R361.